src/bufferutil.cc

```diff
@@ -65,46 +65,90 @@ class BufferUtil : public ObjectWrap
   {
     Nan::HandleScope scope;
     Local<Object> buffer_obj = info[0]->ToObject();
-    size_t length = Buffer::Length(buffer_obj);
     Local<Object> mask_obj = info[1]->ToObject();
-    unsigned int *mask = (unsigned int*)Buffer::Data(mask_obj);
-    unsigned int* from = (unsigned int*)Buffer::Data(buffer_obj);
-    size_t len32 = length / 4;
-    unsigned int i;
-    for (i = 0; i < len32; ++i) *(from + i) ^= *mask;
-    from += i;
-    switch (length % 4) {
-      case 3: *((unsigned char*)from+2) = *((unsigned char*)from+2) ^ ((unsigned char*)mask)[2];
-      case 2: *((unsigned char*)from+1) = *((unsigned char*)from+1) ^ ((unsigned char*)mask)[1];
-      case 1: *((unsigned char*)from  ) = *((unsigned char*)from  ) ^ ((unsigned char*)mask)[0];
-      case 0:;
-    }
+
+    char* mask = Buffer::Data(mask_obj);
+    char* from = Buffer::Data(buffer_obj);
+    size_t length = Buffer::Length(buffer_obj);
+
     info.GetReturnValue().Set(Nan::True());
+
+    /* Alignment preamble */
+    size_t index;
+    for (index = 0; index < length && (reinterpret_cast<size_t>(from) & 0x07); ++index)
+        *from++ ^= mask[index % 4];
+    length -= index;
+    if (!length)
+        return;
+
+    /* Realign mask and convert to 64 bit */
+    char maskAlignedArray[8];
+    for (size_t i = 0; i < 8; i++, index++)
+        maskAlignedArray[i] = mask[index % 4];
```

Comment:
The compiler generates the same output in both cases. `index % 4` is clearer so I prefer that.